AWSKinesis/AWSAbstractKinesisRecorder.m

```diff
@@ -178,17 +178,14 @@ - (AWSTask *)saveRecord:(NSData *)data
                                ];
                 if (!result) {
                     AWSLogError(@"SQLite error. [%@]", db.lastError);
-                    error = db.lastError;
+                    dbError = db.lastError;
                 }
             }];
         }
-
-        if (error) {
```

Comment:
This changes the logical flow of the method. Probably, we should fail fast when there are any issues with SQLite.